File: browser_extension/logger.js
/**
 * Minimaler Logger mit einstellbarem Level, kein Build-System noetig.
 * Wird in jedem Kontext einzeln eingebunden (Content Scripts, Popup,
 * Background), da diese sich keine Variablen teilen.
 */

const LOG_LEVELS = { trace: 0, debug: 1, info: 2, warn: 3, error: 4 };

// Zentral hier einstellen: debug waehrend der Entwicklung, info/warn fuer
// normale Nutzung, um die Konsole ruhig zu halten.
const CURRENT_LOG_LEVEL = LOG_LEVELS.info;

function  logTrace(...args){
    if(CURRENT_LOG_LEVEL <= LOG_LEVELS.trace) console.log("[PII Filter]", ...args);
}
function logDebug(...args) {
  if (CURRENT_LOG_LEVEL <= LOG_LEVELS.debug) console.log("[PII Filter]", ...args);
}
function logInfo(...args) {
  if (CURRENT_LOG_LEVEL <= LOG_LEVELS.info) console.log("[PII Filter]", ...args);
}
function logWarn(...args) {
  if (CURRENT_LOG_LEVEL <= LOG_LEVELS.warn) console.warn("[PII Filter]", ...args);
}
function logError(...args) {
  if (CURRENT_LOG_LEVEL <= LOG_LEVELS.error) console.error("[PII Filter]", ...args);
}

File: host/src/piifilter/logger.py
"""
Minimaler Logger mit einstellbarem Level, spiegelt das Schema von
logger.js (JS-Seite): trace < debug < info < warn < error.

trace/debug duerfen Klartext (Original-PII, Nutzereingaben) enthalten.
info/warn/error NICHT - dort nur Metadaten (Anzahl Funde, Status).
"""
import logging

TRACE_LEVEL = 5
logging.addLevelName(TRACE_LEVEL, "TRACE")


def _trace(self, message, *args, **kwargs):
    if self.isEnabledFor(TRACE_LEVEL):
        self._log(TRACE_LEVEL, message, args, **kwargs)


logging.Logger.trace = _trace

# Zentral hier einstellen: TRACE_LEVEL waehrend der Entwicklung,
# logging.INFO oder hoeher fuer normale Nutzung
CURRENT_LOG_LEVEL = logging.INFO

logging.basicConfig(
    filename="/tmp/pii_filter_host.log",
    level=CURRENT_LOG_LEVEL,
    format="%(asctime)s %(levelname)s %(message)s",
)

# Presidios eigene, sehr gespraechige Logs separat gedaempft,
# unabhaengig vom eigenen Level oben
logging.getLogger("presidio-analyzer").setLevel(logging.WARNING)

log = logging.getLogger("piifilter")